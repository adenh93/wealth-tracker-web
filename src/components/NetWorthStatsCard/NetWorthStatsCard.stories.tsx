import NetWorthStatsCard, { NetWorthStatsCardProps } from './NetWorthStatsCard'
import { Story } from '@storybook/react'

const config = {
  title: 'Components/NetWorthStatsCard',
  component: NetWorthStatsCard,
}

const Template: Story<NetWorthStatsCardProps> = (args) => (
  <NetWorthStatsCard {...args} />
)

export const Primary = Template.bind({})

Primary.args = {
  gain24Hr: 14560,
  netWorth: 975650,
}

export default config
