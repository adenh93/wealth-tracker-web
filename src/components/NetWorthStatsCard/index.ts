export { default } from './NetWorthStatsCard'
export type { NetWorthStatsCardProps } from './NetWorthStatsCard'