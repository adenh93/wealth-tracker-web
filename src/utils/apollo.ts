import { ApolloError } from '@apollo/client'

export const extractInputErrorField = <FormType>(
  error: ApolloError
): keyof FormType | null => {
  if (error.graphQLErrors.length)
    return error.graphQLErrors[0].extensions?.field as keyof FormType
  return null
}
