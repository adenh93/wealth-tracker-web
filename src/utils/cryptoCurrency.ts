import { CMC_LOGO_URL } from '../constants'

export const getCMCLogoUrl = (id: number): string => `${CMC_LOGO_URL}/${id}.png`
