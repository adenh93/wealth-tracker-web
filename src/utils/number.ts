const defaultOpts: Intl.NumberFormatOptions = {
  maximumFractionDigits: 2,
  notation: 'standard',
  signDisplay: 'never',
}

export const formatNumber = (
  value: number,
  opts: Intl.NumberFormatOptions = defaultOpts
): string =>
  new Intl.NumberFormat('en-US', {
    ...opts,
  }).format(value)

export const formatCurrency = (
  value: number,
  opts: Intl.NumberFormatOptions = defaultOpts
): string =>
  formatNumber(value, { ...opts, style: 'currency', currency: 'AUD' })

export const formatPercent = (
  value: number,
  opts: Intl.NumberFormatOptions = defaultOpts
): string => formatNumber(value, { ...opts, style: 'percent' })
