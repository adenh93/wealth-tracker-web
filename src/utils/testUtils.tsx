import { ReactElement } from 'react'
import { ThemeProvider } from '@mui/material'
import { render as tlRender, RenderResult } from '@testing-library/react'
import theme from '../theme'

export const render = (component: ReactElement): RenderResult =>
  tlRender(<ThemeProvider theme={theme}>{component}</ThemeProvider>)
