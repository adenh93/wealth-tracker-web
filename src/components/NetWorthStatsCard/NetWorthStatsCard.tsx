import { FC } from 'react'
import { Typography } from '@mui/material'
import { green, lightGreen } from '@mui/material/colors'
import AttachMoneyIcon from '@mui/icons-material/AttachMoney'
import StatsCard from '../ui/StatsCard'
import ProfitLossLabel from '../ProfitLossLabel'
import { formatCurrency } from '../../utils/number'

export interface NetWorthStatsCardProps {
  gain24Hr: number
  netWorth: number
}

const NetWorthStatsCard: FC<NetWorthStatsCardProps> = ({
  gain24Hr,
  netWorth,
}) => (
  <StatsCard
    Icon={AttachMoneyIcon}
    iconColor={green[500]}
    iconBackgroundColor={lightGreen[100]}
  >
    <ProfitLossLabel value={gain24Hr} type="currency" variant="subtitle1" />
    <Typography variant="body1">{formatCurrency(netWorth)}</Typography>
    <Typography variant="body2">Total Net Worth</Typography>
  </StatsCard>
)

export default NetWorthStatsCard
