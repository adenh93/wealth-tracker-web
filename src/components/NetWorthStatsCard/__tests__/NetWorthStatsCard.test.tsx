import { render } from '../../../utils/testUtils'
import NetWorthStatsCard from '../NetWorthStatsCard'

describe('NetWorthStatsCard component tests', () => {
  test('matches snapshot', () => {
    const { asFragment } = render(
      <NetWorthStatsCard gain24Hr={9999.99} netWorth={99999.99} />
    )

    expect(asFragment).toMatchSnapshot()
  })

  test('it shows positive gains', () => {
    const { getByText } = render(
      <NetWorthStatsCard gain24Hr={9999.99} netWorth={99999.99} />
    )

    expect(getByText('+ A$9,999.99')).toMatchSnapshot()
    expect(getByText('A$99,999.99')).toMatchSnapshot()
  })

  test('it shows negative gains', () => {
    const { getByText } = render(
      <NetWorthStatsCard gain24Hr={-9999.99} netWorth={-99999.99} />
    )

    expect(getByText('- A$9,999.99')).toMatchSnapshot()
    expect(getByText('A$99,999.99')).toMatchSnapshot()
  })
})
