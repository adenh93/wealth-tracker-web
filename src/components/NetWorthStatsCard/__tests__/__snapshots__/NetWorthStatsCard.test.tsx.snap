// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetWorthStatsCard component tests it shows negative gains 1`] = `
<p
  class="MuiTypography-root MuiTypography-subtitle1 css-y57afs-MuiTypography-root"
>
  -
   
  A$9,999.99
</p>
`;

exports[`NetWorthStatsCard component tests it shows negative gains 2`] = `
<p
  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
>
  A$99,999.99
</p>
`;

exports[`NetWorthStatsCard component tests it shows positive gains 1`] = `
<p
  class="MuiTypography-root MuiTypography-subtitle1 css-14hiap2-MuiTypography-root"
>
  +
   
  A$9,999.99
</p>
`;

exports[`NetWorthStatsCard component tests it shows positive gains 2`] = `
<p
  class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
>
  A$99,999.99
</p>
`;

exports[`NetWorthStatsCard component tests matches snapshot 1`] = `[Function]`;
